tas/views.py

```diff
@@ -79,17 +92,22 @@ def profile_edit(request):
     if request.method == 'POST':
         request_pi_eligibility = request.POST.get('request_pi_eligibility')
         form = UserProfileForm(request.POST, initial=tas_user)
+        '''
+            if user is federated only check/update PI Eligibility Requests
+        '''
+        if is_federated and request_pi_eligibility:
+            pie_request = PIEligibility()
+            pie_request.requestor = request.user
+            pie_request.save()
+            _create_ticket_for_pi_request(tas_user, is_federated)
```

Comment:
What does this function do for federated users? What if they're not in TAS?

Reply:
This creates a pi eligibility request in django, then creates a ticket for the request. The ticket create function passes in tas_user but that var is pre-populated with non-tas data if the user is federated, could update the name though.

```    if is_federated:
        tas_user = {}
        tas_user['firstName'] = request.user.first_name
        tas_user['lastName'] = request.user.last_name
        tas_user['email'] = request.user.email```